mediapipe/calculators/instantmotiontracking/model_matrix_manager_calculator.cc

```diff
@@ -0,0 +1,154 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <memory>
+
+#include "Eigen/Dense"
+#include "Eigen/src/Core/util/Constants.h"
+#include "Eigen/src/Geometry/Quaternion.h"
+#include "absl/memory/memory.h"
+#include "absl/strings/str_cat.h"
+#include "absl/strings/str_join.h"
+#include "mediapipe/framework/calculator_framework.h"
+#include "mediapipe/framework/port/ret_check.h"
+#include "mediapipe/framework/port/status.h"
+#include "mediapipe/graphs/object_detection_3d/calculators/box.h"
+#include "mediapipe/graphs/object_detection_3d/calculators/model_matrix.pb.h"
+#include <cmath>
+#include "sticker.h"
+
+namespace mediapipe {
+  namespace{
+    using Matrix4fRM = Eigen::Matrix<float, 4, 4, Eigen::RowMajor>;
+  }
+
+// Intermediary for sticker data to model matrix usable by gl_animation_overlay_calculator
+// *The structure of process() is largely dependent on the rendering system format
+//
+// Input:
+//  STICKERS - Vector of Sticker objects to determine rendering
+//  IMU_DATA - float array of radians [roll, pitch, yaw] from device orientation sensors
+// Output:
+//  MODEL_MATRIX_FLOAT - Result model matrix, in float array form
+//
+// Example config:
+//node{
+//  calculator: "ModelMatrixManagerCalculator"
+//  input_stream: "STICKERS:sticker_data"
+//  input_stream: "IMU_DATA:imu_data"
+//  output_stream: "MODEL_MATRIX_FLOAT:model_matrix"
+//}
+//
+
+class ModelMatrixManagerCalculator : public CalculatorBase {
+ public:
+  ModelMatrixManagerCalculator() {}
+  ~ModelMatrixManagerCalculator() override {}
+  ModelMatrixManagerCalculator(
+      const ModelMatrixManagerCalculator&) = delete;
+  ModelMatrixManagerCalculator& operator=(
+      const ModelMatrixManagerCalculator&) = delete;
+
+  static ::mediapipe::Status GetContract(CalculatorContract* cc);
+
+  ::mediapipe::Status Open(CalculatorContext* cc) override;
+
+  ::mediapipe::Status Process(CalculatorContext* cc) override;
+
+  void setObjectRotation(float roll, float pitch, float yaw, float userRotation);
```

Comment:
naming for functions are: SetObjectRotation
naming for variables are: use_rotation.